From f1a5f20d907681c877843083b9830eafdf2d9b8e Mon Sep 17 00:00:00 2001
From: Vadim Pasternak <vadimp@nvidia.com>
Date: Thu, 7 Oct 2021 19:30:05 +0000
Subject: [PATCH backport 4.19/Alligator 9/9] mlxsw: core: Extend external
 cooling device whitelist for thermal zone binding

Allow thermal zone binding to an external cooling device of type
"emc2305".
Motivation to support new system SN2021 equipped with ASIC Spectrum-1
supporting 48x Gbps RJ45 + 4x100G QSFP28 ports.
System airflow control is provided by EMC2305 RPM-based PWM Fan Speed
Controller as colling device.

Signed-off-by: Vadim Pasternak <vadimp@nvidia.com>
---
 drivers/net/ethernet/mellanox/mlxsw/core_thermal.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/mellanox/mlxsw/core_thermal.c b/drivers/net/ethernet/mellanox/mlxsw/core_thermal.c
index a1a5ef04e..1c429796c 100644
--- a/drivers/net/ethernet/mellanox/mlxsw/core_thermal.c
+++ b/drivers/net/ethernet/mellanox/mlxsw/core_thermal.c
@@ -41,6 +41,7 @@
 /* External cooling devices, allowed for binding to mlxsw thermal zones. */
 static char * const mlxsw_thermal_external_allowed_cdev[] = {
 	"mlxreg_fan",
+	"emc2305",
 };
 
 enum mlxsw_thermal_trips {
-- 
2.20.1

